benchmarks/fuzzy_join_full_benchmark.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from sklearn.metrics.pairwise import distance_metrics
+from sklearn.preprocessing import StandardScaler
+import fasttext.util
+from fasttext import load_model
+from thefuzz.fuzz import partial_ratio, WRatio, ratio
+from thefuzz import process
+from autofj import AutoFJ
+from dirty_cat._fuzzy_join import fuzzy_join
+from thefuzz.fuzz import partial_ratio, WRatio, ratio
+import fasttext.util
+from fuzzy_join_benchmark import fetch_data, fuzzy_join_precision_recall,\
+    fasttext_precision_recall,thefuzz_precision_recall, autofj_precision_recall
+
+
+if __name__ == "__main__":
+    datasets = [
+        "Country",
+        "BasketballTeam",
+        "Drug",
+        "Device",
+        "ArtificialSatellite",
+        "Amphibian",
+        "Song",
+        "HistoricBuilding",
+        "Wrestler",
+        "EthnicGroup",
+    ]
+    df = pd.DataFrame()
+    for dataset in datasets:
+        left_1, right_1, gt_1 = fetch_data(dataset)
+        for analyser in ["word", "char", "char_wb"]:
+            for max_n_gram in [3, 4]:
+                for similarity in ["cosine", "l1", "l2"]:
```

Comment:
Would `itertools.product` be useful here, flattening loops and removing indentation levels?

```python
for analyser, max_n_gram, similarity in product(["word", "char", "char_wb"], [3, 4], ["cosine", "l1", "l2"]):
    # ...
```